packages/vhd-cli/src/commands/compare.js

```diff
@@ -0,0 +1,80 @@
+import { getSyncedHandler } from '@xen-orchestra/fs'
+import { openVhd, Constants } from 'vhd-lib'
+import { resolve } from 'path'
+import Disposable from 'promise-toolbox/Disposable'
+
+const deepCompareObjects = function (src, dest, path) {
+  for (const key of Object.keys(src)) {
+    const srcValue = src[key]
+    const destValue = dest[key]
+    if (srcValue !== destValue) {
+      const srcType = typeof srcValue
+      const destType = typeof destValue
+      if (srcType !== destType) {
+        throw new Error(`key ${path + '/' + key} is of type *${srcType}* in source and *${destType}* in dest`)
+      }
+
+      if (srcType !== 'object') {
+        throw new Error(`key ${path + '/' + key} is *${srcValue}* in source and *${destValue}* in dest`)
+      }
+
+      if (Buffer.isBuffer(srcValue)) {
+        if (!(Buffer.isBuffer(destValue) && srcValue.equals(destValue))) {
+          throw new Error(`key ${path + '/' + key} is buffer in source that does not equal dest`)
+        }
+      } else {
+        deepCompareObjects(src[key], dest[key], path + '/' + key)
+      }
+    }
+  }
+}
+
+export default async args => {
+  if (args.length < 2 || args.some(_ => _ === '-h' || _ === '--help')) {
+    return `Usage: compare <source VHD> <destination> `
+  }
+  const [sourcePath, destPath] = args
+
+  await Disposable.use(async function* () {
+    const handler = yield getSyncedHandler({ url: 'file:///' })
+    const src = yield openVhd(handler, resolve(sourcePath))
+    const dest = yield openVhd(handler, resolve(destPath))
+
+    // parent locator entries contains offset that can be different without impacting the vhd
+    // we'll compare them later
+    const { parentLocatorEntry: srcParentLocatorEntry,tableOffset: srcTableOffset, checksum: srcChecksum,  ...srcHeaderWithoutOffsets } = src.header
+    const { parentLocatorEntry: destParentLocatorEntry,tableOffset: destTableOffset, checksum: destChecksum,  ...destHeaderWithoutOffsets } = dest.header
+    deepCompareObjects(srcHeaderWithoutOffsets, destHeaderWithoutOffsets, 'header')
```

Comment:
Use `lodash/omit` instead:

```suggestion
    const ignoredEntries= [ 'checksum', 'parentLocatorEntry', 'tableOffset' ]
    deepCompareObjects(
      omit(src.header, ignoredEntries),
      omit(dest.header, ignoredEntries)
    )
```